crates/ra_ide/src/syntax_highlighting/injection.rs

```diff
@@ -0,0 +1,169 @@
+//! Syntax highlighting injections such as highlighting of documentation tests.
+
+use super::HighlightedRangeStack;
+use crate::{call_info::ActiveParameter, Analysis, HighlightTag, HighlightedRange, RootDatabase};
+use ast::{HasQuotes, HasStringValue};
+use hir::Semantics;
+use ra_syntax::{ast, AstToken, SyntaxElement, SyntaxToken, TextRange, TextSize};
+use std::{collections::BTreeMap, convert::TryFrom};
+use stdx::SepBy;
```

Comment:
https://github.com/rust-analyzer/rust-analyzer/tree/master/docs/dev#order-of-imports